core/src/wsv.rs

```diff
@@ -560,11 +564,9 @@ impl WorldStateView {
         &self,
         id: &AccountId,
         f: impl FnOnce(&Account) -> T,
-    ) -> Result<T, FindError> {
+    ) -> Result<T, QueryError> {
         let domain = self.domain(&id.domain_id)?;
-        let account = domain
-            .account(id)
-            .ok_or_else(|| FindError::Account(id.clone()))?;
+        let account = domain.account(id).ok_or(QueryError::Unauthorized)?;
```

Comment:
it looks strange to me that not being able to find the account means the account is not authorized.